core/deployment/src/main/java/io/quarkus/deployment/dev/ClassComparisonUtil.java

```diff
@@ -19,6 +21,13 @@
 import org.jboss.jandex.Type;
 
 public class ClassComparisonUtil {
+    static Set<DotName> ignores = new HashSet<>();
+
+    static {
+        ignores.add(DotName.createSimple("kotlin.jvm.internal.SourceDebugExtension"));
+        ignores.add(DotName.createSimple("kotlin.Metadata"));
+    }
+
```

Comment:
This could be simplified to the following and made private (and you would need to adjust the name below):

```suggestion
    private static final Set<DotName> IGNORED_ANNOTATIONS = Set.of(DotName.createSimple("kotlin.jvm.internal.SourceDebugExtension"), DotName.createSimple("kotlin.Metadata"));

```